xmipptomo/protocols/protocol_deep_misalignment_detection.py

```diff
@@ -0,0 +1,423 @@
+    """
+
+    _label = 'detect misalignment from fiducials'
+    _devStatus = BETA
+    _conda_env = 'xmipp_DLTK_v1.0'
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.alignedTomograms = None
+        self.strongMisalignedTomograms = None
+        self.weakMisalignedTomograms = None
+        self.outputSubtomos = None
+        self.isot = None
+
+    # --------------------------- DEFINE param functions ------------------------
+    def _defineParams(self, form):
+        form.addSection(label='Input')
+
+        form.addParam('inputSetOfCoordinates',
+                      PointerParam,
+                      pointerClass=SetOfCoordinates3D,
+                      label='Fiducial 3D coordinates',
+                      help='3D coordinates indicating the location of the fiducials (gold beads) in the tomogram. '
+                           'These fiducails will be the ones used to study misalignment artifacts over them. '
+                           'The coordinate denotes the center of the subtomogram')
+
+        form.addParam('tomoSource',
+                      EnumParam,
+                      choices=['same as picking', 'other'],
+                      default=0,
+                      display=EnumParam.DISPLAY_HLIST,
+                      label='Tomogram source',
+                      help='By default the subtomograms will be extracted from the tomogram used in the picking step '
+                           '( _same as picking_ option ). \n'
+                           'If you select _other_ option, you must provide a different tomogram to extract from. \n'
+                           '*Note*: In the _other_ case, ensure that provided tomogram and coordinates are related ')
+
+        form.addParam('inputSetOfTomograms',
+                      PointerParam,
+                      pointerClass=SetOfTomograms,
+                      condition='tomoSource==1',
+                      allowsNull=True,
+                      label='Input tomograms',
+                      help='Tomograms from which extract the fiducials (gold beads) at the specified coordinates '
+                           'locations.')
+
+        form.addParam('misaliThrBool',
+                      BooleanParam,
+                      default=True,
+                      label='Use misalignment threshold?',
+                      help='Threshold to settle if a tomogram presents weak or strong misalignment. If this value is '
+                           'not provided two output set of tomograms are generated, those discarded which present '
+                           'strong misalignment and those which do not. If this value is provided the second group of '
+                           'tomograms is splitted into two, using this threshold to settle if the tomograms present'
+                           'or not a weak misalignment.')
+
+        form.addParam('misaliThr',
```

Comment:
Units?

Reply:
It is an score parameter, i added the range value